content-src/components/NewTabPage/NewTabPage.js

```diff
@@ -37,10 +38,10 @@ const NewTabPage = React.createClass({
     document.title = "New Tab";
     setFavicon("newtab-icon.svg");
   },
-  componentWillReceiveProps(nextProps) {
-    if (nextProps.isReady && !this.state.renderedOnce) {
+  componentDidUpdate(nextProps) {
+    if (!renderedOnce && nextProps.isReady) {
```

Comment:
Lol 😜  good catch.. totally derp on my part

`isReady` is set by the new tab selector, which is right here:

https://github.com/mozilla/activity-stream/blob/master/content-src/selectors/selectors.js#L107

It will get triggered when "init" is true for all of these things